Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/text/TextResource.kt

```diff
@@ -0,0 +1,33 @@
+package de.rki.coronawarnapp.ccl.dccwalletinfo.text
+
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.CCLText
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.Parameters
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.PluralText
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.SingleText
+import java.util.Locale
+
+/**
+ * Formats [CCLText] lazily when accessed
+ */
+fun textResource(
+    cclText: CCLText,
+    locale: Locale = Locale.getDefault()
+) = lazy { formatCCLText(cclText, locale.language) }
+
+internal fun formatCCLText(cclText: CCLText, lang: String): String = when (cclText) {
+    is PluralText -> "TODO: Format plural text"
+    is SingleText -> {
+        // DO: provide fallback
+        val text = cclText.localizedText[lang]
```

Comment:
We should consider the list of languages from Android, otherwise we might get a mix of languages, eg, when the first language is Spanish, the second is Bulgarian

Reply:
never mind, that will be handled in the dcc layer